hydrosdk/predictor.py

```diff
@@ -83,31 +83,30 @@ def predict_resp_to_dict_pydtype(response: PredictResponse) -> dict:
 
             dims = [dim.size for dim in tensor_shape.dim]
 
-            output_tensors_dict[key_str] = dims
+            # FIXME value is not extracted from tensor_proto, see  predict_resp_to_dict_nparray
+            output_tensors_dict[key_str] = dims  # FIXME dims is not a value! it is dimenshion shape
         return output_tensors_dict
 
     @staticmethod
     def predict_resp_to_dict_nparray(response: PredictResponse) -> dict:
         output_tensors_dict = {}
-        for key, value in response.outputs.items():
-            key_str = key
-
-            tensor_shape = value.tensor_shape
-
-            dims = [dim.size for dim in tensor_shape.dim]
-
-            dtype = proto2np_dtype(value.dtype)
-            output_tensors_dict[key_str] = np.asarray(dims, dtype=dtype)
+        for tensor_name, tensor_proto in response.outputs.items():
+            array_shape = [dim.size for dim in tensor_proto.tensor_shape.dim]
+            np_dtype = proto2np_dtype(tensor_proto.dtype)
+            value = getattr(tensor_proto, DTYPE_TO_FIELDNAME[tensor_proto.dtype])
+            output_tensors_dict[tensor_name] = np.asarray(value, dtype=np_dtype).reshape(*array_shape if array_shape else (-1,))
 
         return output_tensors_dict
 
     @staticmethod
     def predict_resp_to_df(response: PredictResponse) -> pd.DataFrame:
         df = pd.DataFrame()
-        for key, value in response.outputs.items():
+        for key, tensor_proto in response.outputs.items():
             current_index = key
-            tensor_shape = value.tensor_shape
+            tensor_shape = tensor_proto.tensor_shape
 
+            # FIXME create pd.Dataframe, not pd.Series. reuse code from predict_resp_to_dict_nparray?
+            # FIXME value is not extracted from tensor_proto, see  predict_resp_to_dict_nparray
```

Comment:
my tests extracted the values. Could you provide the code you used?

Reply:
Values extracted are tensor dimensions, not tensor values.

By running `dims = pd.Series([dim.size for dim in tensor_shape.dim])` you create pd.Series with the shape of the tensor. Values of TensorProto are stored in attributes like 'double_val', 'in64_val' etc, depending on data type used. These attributes were not previously accessed in the `predict_resp_to_dict_nparray`, so that's what I meant by not extracting value.

It seems that tests do not check exact returned value, they only check class of returned value. i.e. for numpy conversions test looks like 
```python
    assert isinstance(predictions['out1'], np.ndarray)
    assert isinstance(predictions, dict)
```

In the case of tensors, your code returns the shape of a tensor, in an np.array, so the test is passed, though values returned are wrong

In the case of scalar, your code will return an empty np.array, so the test still passes, though values returned are still wrong

We definitely need tests that will check the correctness of values converted to and from proto, wdy?

I'll make one in the meantime